### PR TITLE
DEX-1821 Dummy APMs

### PR DESCRIPTION
**Pull request by d4r1091**
# What this PR does

- Implemented DummyAPMs with UI Tests
- Minor codebase improvements

### CODE REVIEW

File: Sources/PrimerSDK/Classes/User Interface/Root/PrimerRootViewController.swift [hunk 1]
```diff
@@ -490,7 +490,13 @@ internal class PrimerRootViewController: PrimerViewController {
             return viewController.view.bounds.size.height + nc.navigationBar.bounds.height
         }
     }
+}
+
+extension PrimerRootViewController {
     
+    func presentPrimerTestViewControllerForPaymentMethodType(_ type: PrimerPaymentMethodType) {
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/User Interface/Root/PrimerRootViewController.swift:497]: Why is this needed?

File: Sources/PrimerSDK/Classes/User Interface/TestPaymentMethods/FlowDecisionTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        
+        self.preservesSuperviewLayoutMargins = false
+        self.contentView.preservesSuperviewLayoutMargins = false
+        self.selectionStyle = .none
+        
+        let theme: PrimerThemeProtocol = DependencyContainer.resolve()
+        backgroundColor = theme.view.backgroundColor
+        layer.cornerRadius = 8
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/User Interface/TestPaymentMethods/FlowDecisionTableViewCell.swift:27]: I believe that the current structure of theme contains PMs corner radius

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+            tableView.contentInsetAdjustmentBehavior = .never
+        }        
+        tableView.register(FlowDecisionTableViewCell.self, forCellReuseIdentifier: FlowDecisionTableViewCell.identifier)
+        tableView.register(HeaderFooterLabelView.self, forHeaderFooterViewReuseIdentifier: "header")
+        tableView.dataSource = self
+        tableView.delegate = self
+        return tableView
+    }()
+    
+    private lazy var _originalImage: UIImage? = {
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift:44]: Why don't you set them in `PaymentMethodTokenizationViewModel` as you did with Klarna?

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+        
+        self.didStartPayment = {
+            self.submitButton.startAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = false
+        }
+        
+        self.didFinishPayment = { err in
+            self.submitButton.stopAnimating()
+            Primer.shared.primerRootVC?.view.isUserInteractionEnabled = true
+            self.willDismissPaymentMethodUI?()
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift:115]: Should they be called at the same time? Both `present` and `push` provide a finish callback to call the `didDismiss`.

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+            self.didDismissPaymentMethodUI?()
+        }
+        
+        super.start()
+    }
+    
+    override func startTokenizationFlow() -> Promise<PrimerPaymentMethodTokenData> {
+        
+        DispatchQueue.main.async {
+            UIApplication.shared.endIgnoringInteractionEvents()
```
Comment (EvansPie) [inline Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift:125]: You probably want to start ignoring here, right? Also I recently made a change and I always call this in `ensure`.

File: Sources/PrimerSDK/Classes/User Interface/Root/PrimerRootViewController.swift (hunk 1 above)
Comment (d4r1091) [reply Sources/PrimerSDK/Classes/User Interface/Root/PrimerRootViewController.swift:497]: Not needed - leftover from the implementation I thought it would be good with HUC - then realized not needed

File: Sources/PrimerSDK/Classes/User Interface/TestPaymentMethods/FlowDecisionTableViewCell.swift (hunk 2 above)
Comment (d4r1091) [reply Sources/PrimerSDK/Classes/User Interface/TestPaymentMethods/FlowDecisionTableViewCell.swift:27]: That corner radius is for the decision cells "Authorized / Declined / Fail". PM corner radius hasn't been touched 👍

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift (hunk 3 above)
Comment (d4r1091) [reply Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift:44]: Good point. The reason why I decided to use the `originalImage` instead of the `buttonImage` is due to the fact that `originalImage` is utilized for the navigation bar logo in the flow decision view. 
The navigation bar has no background color there so I needed different variations of the PM logos.

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift (hunk 4 above)
Comment (d4r1091) [reply Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/PrimerTestPaymentMethodTokenizationViewModel.swift:115]: I've updated it @EvansPie - let me know if now makes more sense 🙏 
Screen Shot 2022-06-06 at 14 42 18
